idaes/core/solvers/petsc.py

```diff
@@ -0,0 +1,626 @@
+            res_list.append(res)
+    return res_list
+
+
+class PetscTrajectory(object):
+    def __init__(
+            self,
+            stub,
+            pth=None,
+            vis_dir="Visualization-data",
+            delete_on_read=False,
+            ):
+        """Class to read PETSc TS solver trajectory data.
+
+        Args:
+            stub (str): file name stub for variable info
+            pth (str): path where variable info and trajectory data are stored
+                if None, use current working directory
+            vis_dir (str): subdirectory where visualization data is stored
+            delete_on_read (bool): if true delete trajectory data after reading
+        """
+        if PetscBinaryIOTrajectory is None:
+            raise RuntimeError("PetscBinaryIOTrajectory could not be imported")
+        if pth is not None:
+            stub = os.path.join(pth, stub)
+            vis_dir = os.path.join(pth, vis_dir)
+        self.stub = stub
+        self.vis_dir = vis_dir
+        self.path = pth
+        self._read()
+        if delete_on_read:
+            self.delete_files()
+
+    def _read(self):
+        with open(f'{self.stub}.col') as f:
+            names = list(map(str.strip, f.readlines()))
+        with open(f'{self.stub}.typ') as f:
+            typ = list(map(int,f.readlines()))
+        self.vars = [name for i, name in enumerate(names) if typ[i] in [0,1]]
+        (t, v, names) = PetscBinaryIOTrajectory.ReadTrajectory(
+            "Visualization-data"
+        )
+        self.time = t
+        self.vecs_by_time = v
+        self.vecs = dict.fromkeys(self.vars, None)
+        for k in self.vecs.keys():
+            self.vecs[k] = [0]*len(self.time)
+        self.vecs["_time"] = list(self.time)
+        for i, v in enumerate(self.vars):
+            for j, vt in enumerate(self.vecs_by_time):
+                self.vecs[v][j] = vt[i]
+
+    def _var_name_list(self, vars):
+        if isinstance(vars, str):
+            vars = [vars]
+        if hasattr(vars, "ctype"):
+            if issubclass(vars.ctype, pyo.Var):
+                vars = [vars]
+        vars = list(map(str, vars))
+        for name in vars:
+            if name not in self.vars:
+                raise RuntimeError(f"Variable {name} not found.")
+        return vars
+
+    def get_vec(self, var):
+        """Return the time vector for var.
+
+        Args:
+            var (str or Var): Variable to get vector for.
+
+        Retruns:
+            (list): time vector
+
+        """
+        var = str(var)
+        return self.vecs[var]
+
+    def get_dt(self):
+        """Get a list of time steps
+
+        Args:
+            None
+
+        Returns:
+            (list)
+        """
+        dt = [None]*(len(self.time) - 1)
+        for i in range(len(self.time)-1):
+            dt[i] = self.time[i + 1] - self.time[i]
+        return dt
+
+    def interpolate_vecs(self, times):
+        """Create a new vector dictionary interpolated at times.
```

Comment:
Maybe mention that this method does in fact extrapolate if a time is outside the range of times in self.times?